### PR TITLE
ajd chapter 17

### PR DESCRIPTION
**Pull request by alexdeas**
ajd chapter 17

### CODE REVIEW

File: answers.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+1) ``javafx``
+
+2) ``Stage`` and ``Scene``? Seems too obvious.
```
Comment (utopia-2) [inline answers.md:3]: Sounds like a _good_ thing for an API :boar:.

File: src/chapter17/EffectsAndTransformsDemo.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  
+  // Blur the rotate button too
+  BoxBlur rotateBlur = new BoxBlur(5.0, 5.0, 2); 
+  boolean doBlur = true;
+ 
+  public static void main(String[] args) {  
+  
+    // Start the JavaFX application by calling launch().  
+    launch(args);    
+  }
```
Comment (utopia-2) [inline src/chapter17/EffectsAndTransformsDemo.java:40]: The latest eclipse seems to have run configurations for JavaFX applications (so you wouldn't need `main`).

File: src/chapter17/JavaFXEventDemo.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+import javafx.application.*; 
+import javafx.scene.*; 
+import javafx.scene.layout.*; 
+import javafx.scene.control.*; 
+import javafx.geometry.*; 
+ 
+public class JavaFXEventDemo extends Application { 
+ 
+  Label response;
```
Comment (utopia-2) [inline src/chapter17/JavaFXEventDemo.java:15]: This needn't be an instance variable. :crocodile:

File: src/chapter17/JavaFXEventDemo.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+ 
+    // Create two push buttons. 
+    Button btnUp = new Button("Run"); 
+    Button btnDown = new Button("Exit"); 
+ 
+    // Handle the action events for the Run button. 
+    btnUp.setOnAction( (ae) -> response.setText( "You pressed Run" ) ); 
+ 
+    // Handle the action events for the Exit button. 
+    btnDown.setOnAction( (ae) -> Platform.exit() );
```
Comment (utopia-2) [inline src/chapter17/JavaFXEventDemo.java:53]: For single arg lambdas, you can leave out the parentheses, if you like (or if you're in some kind of parenthesis famine): :cold_sweat: 

``` java
ae -> Platform.exit()
```

File: src/chapter17/JavaFXFC.java [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+         if(i != j)
+           result.setText("Files are not the same.");
+         else
+           result.setText("Files are the same.");
+      } catch(IOException exc) {
+        result.setText("File Error");
+      }
+        
+      
+    } );
```
Comment (utopia-2) [inline src/chapter17/JavaFXFC.java:90]: This seems like a lot of body for one lambda - might be worth pulling it out to a method. You lose the advantage of lexical closure capture, but you could make it non-gui specific:

``` java
public String compareFiles(String patha, String pathb);
```

``` java
compare.setOnAction( e -> result.setText(compareFiles(file1.getText(), file2.getText())) );
```

File: answers.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+12) False!
+
+13) See JavaFXFC.java
+
+14) See EffectsAndTransformsDemo.java
+
+15) Ran out of time
+
+16) TBC
```
Comment (utopia-2) [inline answers.md:31]: :shipit:

File: src/chapter17/EffectsAndTransformsDemo.java (hunk 2 above)
Comment (alexdeas) [inline src/chapter17/EffectsAndTransformsDemo.java:40]: That's what I get from copying the example code.

File: src/chapter17/JavaFXEventDemo.java (hunk 4 above)
Comment (alexdeas) [inline src/chapter17/JavaFXEventDemo.java:53]: What if I'm into superfluous parentheses?

File: src/chapter17/JavaFXFC.java (hunk 5 above)
Comment (alexdeas) [inline src/chapter17/JavaFXFC.java:90]: I had a method originally, but took it out as I was only using it once. Hate having single use methods.

Comment (utopia-2) [inline src/chapter17/JavaFXFC.java:90]: Whaaat, a well named single use method that does one conceptual task is far better than a wall of code that interleaves behavior. That said, a single-use _public_ method would be awful.

File: src/chapter17/JavaFXEventDemo.java (hunk 4 above)
Comment (utopia-2) [inline src/chapter17/JavaFXEventDemo.java:53]: Sure, if you're into it, who am I to judge? :kissing_heart:

**Pull request closed without merging**